crates/router/src/services/api.rs

```diff
@@ -278,6 +282,15 @@ where
 {
     // If needed add an error stack as follows
     // connector_integration.build_request(req).attach_printable("Failed to build request");
+    logger::debug!(connector_request=?connector_request);
+    let masked_conn_req = connector_request.as_ref().map(|req| match &req.body {
+        Some(RequestContent::Json(payload))
+        | Some(RequestContent::FormUrlEncoded(payload))
+        | Some(RequestContent::Xml(payload)) => payload.masked_serialize().unwrap_or_default(),
+        _ => serde_json::Value::Null,
```

Comment:
If any new enum variant added we will never know changes required here. So let's not use default, instead let's handle case by case.